follow_sponsorship/view/contract_view.xml

```diff
@@ -0,0 +1,109 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+    Copyright (C) 2014 Compassion (http://www.compassion.ch)
+    @author David Coninckx <david@coninckx.com>
+    The licence is in the file __openerp__.py
+-->
+<openerp>
+	<data>
+        <record id="view_follow_contract_form_compassion" model="ir.ui.view">
+			<field name="name">follow.contract.compassion.form</field>
+			<field name="model">recurring.contract</field>
+			<field name="arch" type="xml">
+                <form string="Follow contract" version="7.0" >
+                    <header>
+                        <field name="sds_state" widget="statusbar" statusbar_visible="draft,start,waiting_welcome,active,cancelled" />
+                        <button name="contract_validated" class="oe_highlight" attrs="{'invisible':[('sds_state','!=','draft')]}" string="Validate"/>
+                        <button name="%(sponsorship_compassion.action_end_sponsorship)d" class="oe_highlight" string="Terminate" states="waiting,mandate,active" type="action"/>
+                        <button name="mail_sent" class="oe_highlight" string="Mail sent" attrs="{'invisible':[('sds_state','!=','start')]}"/>
+                        <button name="welcome_sent" class="oe_highlight" string="Welcome sent" attrs="{'invisible':[('sds_state','!=','waiting_welcome')]}"/>
+                        <button name="no_sub" class="oe_highlight" string="No sub" attrs="{'invisible':[('sds_state','!=','sub_waiting')]}"/>
+                    </header>
+                    <sheet>
+                        <group>
+                            <group>
+                                <field name="partner_id"/>
+                                <field name="correspondant_id"/>
+                                <field name="parent_id"/>
+                            </group>
+                            <group>
+                                <field name="origin_id"/>
+                                <field name="channel"/>
+                            </group>
+
+                            <group>
+                                <field name="start_date"/>
+                                <field name="activation_date"/>
+                                <field name="date_welcome"/>
+                                <field name="end_date"/>
+                            </group>
+                            <group>
+                                <field name="child_id"/>
+                                <field name="end_reason"/>
+                                <field name="state"/>
+                            </group>
+                        </group>
+                    </sheet>
+                </form>
+			</field>
+		</record>
+
+        <record id="view_follow_contract_tree_compassion" model="ir.ui.view">
+			<field name="name">follow.contract.compassion.tree</field>
+			<field name="model">recurring.contract</field>
+			<field name="arch" type="xml">
+                <tree string="Follow contracts" create="false" >

```

Comment:
Put some colors depending on the sds_state. Inspire yourself by contract views and children views as well.